active/Win32/Window_and_Application_API_Spec.md

```diff
@@ -0,0 +1,618 @@
+    - Exception type: TBD
+ 
+    - Exception Message: TBD. Something like "Only one Window per thread is allowed in UWP app model".
+
+### Window.Bounds property
+Gets a Rect value containing the height and width of the application window in logical pixels.
+
+```CS
+public Rect Bounds { get; }
+```
+#### Property value
+
+Rect: A value that reports the height and width of the window.
+
+#### Remarks
+
+- **Win32**: The Window is created with the default size of the Window specify by the Window Shell.
+- **UWP**: Keep the current behavior. To change the size of the CoreWindow in UWP, it’s required to use ApplicationView’s PreferredLaunchViewSize and PreferredLaunchWindowingMode.
+
+### Window.Icon property **[NEW]** 
+Gets or sets a window's icon. For each window, this icon is used in its title bar, its task bar button, and in its ALT+TAB application selection list entry.
+
+```CS
+public ImageSource Icon { get; set; }
+```
+
+#### Property value
+Icon: An ImageSource object that represents the icon.
+
+#### Sample
+```XML
+<Window
+  xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+  xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+  x:Class="Sample.MainWindow"
+    Icon="WinUI3Icon.ico">
+</Window>
+```
+
+#### Remarks:
+- **Win32**: The image is set as an Icon of the Window. 
+- **UWP**: Do nothing. The property value is ignored.
+
+### Window.Content property
+Gets or sets the UIElement visual root of the window. 
+
+```CS
+[ContentProperty(“Content”)]
+public UIElement Content { get; set; }
+```
+#### Property Value
+UIElement: The visual root of a window.
+
+#### Remarks
+This property has the ContentProperty attribute so the XAML designer can display the content on the designer.
+
+### Window.Visible property
+Gets a value that reports whether the window is visible. Visibility does not consider the window being behind another window and not visible. It only detects if it is on the screen. This property matches the XamlRoot’s IsHostVisible property.
+
+```CS
+public bool Visible { get; } 
+```
+
+#### Property Value
+bool: **true** if the window is visible; **false** if the window is not visible.
+
+#### Remarks
+- **Win32**: False when the Window is minimized or never displayed. 
+- **UWP**: Keep the same current UWP behavior.
+
+
+### Window.VisibilityChanged Event
+
+Occurs when the value of the Visible property changes. Also, it’s fired when the XamlRoot’s IsHostVisible changes as a consequence of cascade change of the Window’s Visibility.
+
+```CS 
+Window w = new Window();
+window.Content = new TextBlock() { Text = "Hello" };
+window.VisibilityChanged += OnVisibilityChanged;
+```	
+
+```CS
+private void OnVisibilityChanged(object sender, VisibilityChangedEventArgs e)
+{
+   if (e.Visible)
+   {
+      //
+   }
+}
+```
+#### Remarks
+VisibilityChangedEventArgs includes a Boolean Visible property that you should check to determine the visibility state when the event was fired.
```

Comment:
I'd like to see this entire visibility api moved to UIElement instead. See issue logged here with justifications:
https://github.com/microsoft/microsoft-ui-xaml/issues/674

Reply:
Obvious overlap, but these look like two separate issues. I agree we should have a UIElement.IsVisible property like WPF's, but the Window.Visible property already exists, and Window isn't a UIElement. So we can have both Window.Visible and UIElement.IsVisible.